### PR TITLE
feat(glue): support AWS Glue 5.0

### PR DESCRIPTION
**Pull request by mazyu36**
### Issue # (if applicable)

N/A

### Reason for this change

Glue supported ver 5.0.
Ref: [Introducing AWS Glue 5.0](https://aws.amazon.com/about-aws/whats-new/2024/12/aws-glue-5-0/)

### Description of changes
Add Enum.

### Description of how you validated changes
Update an unit test and an integ test.

### Checklist
- [ ] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by mazyu36** [comment]
** **NOTE** **
This change conflicts with #30833.
I have created this PR to notify about the update, but I am fine with addressing it after the above PR is merged. 
I will leave the decision to the maintainers.

**Review by GavinZZ — APPROVED** [review]

**Pull request merged**
